cluster/addons/metrics-server/metrics-server-deployment.yaml

```diff
@@ -23,24 +23,23 @@ data:
 apiVersion: apps/v1
 kind: Deployment
 metadata:
-  name: metrics-server-v0.3.6
+  name: metrics-server-v0.4.3
   namespace: kube-system
   labels:
     k8s-app: metrics-server
-    kubernetes.io/cluster-service: "true"
     addonmanager.kubernetes.io/mode: Reconcile
-    version: v0.3.6
+    version: v0.4.3
 spec:
   selector:
     matchLabels:
       k8s-app: metrics-server
-      version: v0.3.6
+      version: v0.4.3
```

Comment:
deployment selectors are immutable. this cannot be changed or it will fail to apply updates to existing clusters

Reply:
Update will work as we also change the name of deployment. Instead of doing update on Deployment object, addon-manager will delete the old one and create the new one.

Reply:
ah, I see